react/features/base/media/middleware.js

```diff
@@ -162,7 +162,10 @@ function _setRoom({ dispatch, getState }, next, action) {
     sendAnalytics(createStartAudioOnlyEvent(audioOnly));
     logger.log(`Start audio only set to ${audioOnly.toString()}`);
 
-    dispatch(setAudioOnly(audioOnly));
+    // Set state for audioOnly mode and ensure possibility of creating video
+    // track, only when the room is valid, to avoid creating the track on call
+    // ending.
+    dispatch(setAudioOnly(audioOnly, roomIsValid));
```

Comment:
We should pass `false` here. The `WelcomePage` and `Conference` components already take care of creating the tracks they need.